src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package uk.gov.hmcts.darts.dailylist.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.common.entity.Courthouse;
+import uk.gov.hmcts.darts.common.entity.DailyListEntity;
+import uk.gov.hmcts.darts.courthouse.exception.CourthouseCodeNotMatchException;
+import uk.gov.hmcts.darts.courthouse.exception.CourthouseNameNotFoundException;
+import uk.gov.hmcts.darts.courthouse.service.CourthouseService;
+import uk.gov.hmcts.darts.dailylist.exception.DailyListException;
+import uk.gov.hmcts.darts.dailylist.mapper.DailyListMapper;
+import uk.gov.hmcts.darts.dailylist.model.CourtHouse;
+import uk.gov.hmcts.darts.dailylist.model.DailyList;
+import uk.gov.hmcts.darts.dailylist.model.DailyListPostRequest;
+import uk.gov.hmcts.darts.dailylist.repository.DailyListRepository;
+import uk.gov.hmcts.darts.dailylist.service.DailyListService;
+
+import java.text.MessageFormat;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class DailyListServiceImpl implements DailyListService {
+
+    private final DailyListRepository dailyListRepository;
+    private final CourthouseService courthouseService;
+
+
+    @Override
+    /*
+    Retrieve the new Daily List, and store it in the database.
+     */
+    public void processIncomingDailyList(DailyListPostRequest postRequest) {
+        DailyList dailyList = postRequest.getDailyList();
+
+        dailyList.getCrownCourt().getCourtHouseName();
+        Courthouse courthouse = retrieveCourtHouse(dailyList);
```

Comment:
I thought we had agreed to use an access pattern where by we expose a single point of entry into a feature for use by other features, an interface in an "api" package. As per @chrisbellingham-hmcts presentation a little while ago.

Reply:
i guess in the future, this should be broken up into multiple tickets?
1. Create the new endpoint on Courthouse
2. create a feignClient to call it. (although that quite a small job)
3. create the dailyList service

Reply:
@cmagowan-hmcts not sure I understand. My comment is that we shouldn't be accessing the service within a different "feature" directly. Instead we should access through a single entry point as defined by an interface.

I'm not saying we should use the API endpoint.

Does that make sense?

Reply:
yeah, now implemented